src/Service.Tests/SqlTests/GraphQLMutationTests/DwSqlGraphQLMutationTests.cs

```diff
@@ -434,6 +434,154 @@ public override async Task DeleteMutationWithOnlyTypename()
+            string insertedValue = "{\"id\":1,\"title\":\"Orange Tomato\"}";
+            string expected = await GetDatabaseResultAsync(dbQuery);
+            Assert.IsTrue(actual.ToString().Contains(insertedValue));
+            Assert.IsTrue(expected.ToString().Equals("True"));
+        }
+
+        /// <summary>
+        /// <code>Do: </code>insert new Book and return nothing
+        /// <code>Check: </code>if the intended book is inserted in books table
+        /// </summary>
+        [TestMethod]
+        public async Task TestStoredProcedureMutationForInsertion()
+        {
+            string msSqlQuery = @"
+SELECT CASE 
+         WHEN EXISTS (
+           SELECT 1
+           FROM [books] AS [table0]
+           WHERE [table0].[title] = 'Random Book'
+             AND [table0].[publisher_id] = 1234
+         ) THEN 'true' 
+         ELSE 'false'
+       END AS BookExists
+            ";
+
+            string graphQLMutationName = "executeInsertBook";
+            string graphQLMutation = @"
+                mutation {
+                    executeInsertBook(book_id: 10, title: ""Random Book"", publisher_id: 1234 ) {
+                        result
+                    }
+                }
+            ";
+
+            string currentDbResponse = await GetDatabaseResultAsync(msSqlQuery);
+            Assert.AreEqual(currentDbResponse, "False", "Entry should not exist in the db before insertion");
+            JsonElement graphQLResponse = await ExecuteGraphQLRequestAsync(graphQLMutation, graphQLMutationName, isAuthenticated: true);
+
+            // Stored Procedure didn't return anything
+            SqlTestHelper.PerformTestEqualJsonStrings("[]", graphQLResponse.ToString());
+
+            // check to verify new element is inserted
+            string updatedDbResponse = await GetDatabaseResultAsync(msSqlQuery);
+            Assert.AreEqual(updatedDbResponse, "True", "Entry should be inserted into db.");
+        }
+
+        /// <summary>
+        /// <code>Do: </code>Book title updation and return the updated row
+        /// <code>Check: </code>if the result returned from the mutation is correct and
+        /// updated value exists in the database
+        /// </summary>
+        [TestMethod]
+        public async Task TestStoredProcedureMutationForUpdate()
+        {
+            string dbQuery = @"
+SELECT CASE
+         WHEN EXISTS (
+           SELECT 1
+           FROM [books] AS [table0]
+           WHERE [table0].[id] = 14
+             AND [table0].[publisher_id] = 1234
+             AND [table0].[title] = 'Before Midnight'
+         ) THEN 'true' ELSE 'false'
+       END AS BookExists
+            ";
+
+            string graphQLMutationName = "executeUpdateBookTitle";
+            string graphQLMutation = @"
+                mutation {
+                    executeUpdateBookTitle(id: 14, title: ""Before Midnight"") {
```

Comment:
please remind me.. weren't we not going to allow fields in the respone for DW? and only return a "result" object?

Reply:
That was for DW mutations that are self generated. For stored procedure executions, we will return the result set.